Fix #100 - Docker Error syntax issue

rishabjasrotia opened the pull request:
Fix #100  - Docker Error syntax issue

Outcome: Helps to build the project from root using `docker compose up -d`

- Fixed the environment variable values for local development
- Support for make command: Only need to execute make build
- Support for Local File Storage=> Helps to test File operation in local

andrew-opensignlabs commented:
@rishabjasrotia Great initiative. Thanks for starting this pull request.

I tried testing it from your fork, looks like its conflicting with a cascaded build because of relative path used.
<img width="985" alt="image" src="https://github.com/OpenSignLabs/OpenSign/assets/148278535/091c9523-a851-416f-8998-d590b345e57e">

andrew-opensignlabs commented:
@rishabjasrotia 
I have changed the the server port no to 5001 in order to make it run without any port conflict on Mac.

I think we still need to add default values for env variables or some instructions to the documentation for setting up .env before merging this. Currently all containers are running properly but login/signup is not working without proper env setup.

rishabjasrotia commented:
Yes I agree we can update the documentation for .env

rishabjasrotia commented:
Hello @andrew-opensignlabs,
Starting server with proper env settings is still not documented yet for local development. 
can you let me know what varaiable needed to run opensign server?

rishabjasrotia commented on docker-compose.yml:
```diff
@@ -5,17 +5,17 @@ services:
       - /usr/src/app/node_modules
     ports:
-      - "5000:5000"
+      - "8080:8080"
```
Port 8080 since it the port exposed in OpenSignServer DockerFile

rishabjasrotia commented on docker-compose.yml:
```diff
@@ -44,6 +45,16 @@ services:
       - "3000:3000"
     networks:
       - app-network
+  localstack:
```
Added Localstack to support self hosted AWS S3

andrew-opensignlabs commented:
@rishabjasrotia Great. All containers are starting up with simple "make build" command. 
Frontend and mongodb containers are properly accessible but the server container is somehow not reachable on port 8080 even though docker says its running.

Below is the error I could see in the log -
<img width="1013" alt="image" src="https://github.com/OpenSignLabs/OpenSign/assets/148278535/6b482004-66f7-4385-9692-72f901a582f1">

I am discussing with dev team to see if they can help us resolve it.

rishabjasrotia commented:
Hello @andrew-opensignlabs 
To built new docker images you need to delete the old opensignserver docker images + containers.

No need to connect to dev team.

nxglabs commented:
@rishabjasrotia Appreciate the efforts you are putting.
Undoubtedly this pull req is the biggest value add from the community since launch.

There is one last change we need to do before we merge -
We need to populate the default mongodb collections data from https://github.com/OpenSignLabs/Deploy-OpenSign-to-Digital-Ocean/tree/main/default-data
by running https://github.com/OpenSignLabs/Deploy-OpenSign-to-Digital-Ocean/blob/main/mongo-init.sh
after mongodb container is launched.
Without this default data signup will not work.

rishabjasrotia commented:
Hello @nxglabs @andrew-opensignlabs 

I have added the support to do the initial DB import for mongo.
Now all container are running and signup is working too!

nxglabs commented:
Thanks @rishabjasrotia 
Signup, login and most other functionalities are working properly.
I was testing the file upload functionality and got below error -

**Inaccessible host: `my-bucket.s3.localhost.localstack.cloud' at port `4566'. This service may not be available in the `eu-central-1' region.**

Am i missing anything?

rishabjasrotia commented:
**Yes I had similar issue may be need to check with localhost issue of S3 adapter support to localstack.**

If we use AWS S3 actual bucket setting then it is working.

rishabjasrotia commented:
Hello @nxglabs @andrew-opensignlabs 

**I have removed localstack support for while and added the Support for Local File Storage=> Helps to test File operation in local**

Please have review?
